src/main/cpp/src/parse_uri.cu

```diff
@@ -572,6 +621,20 @@ uri_parts __device__ validate_uri(const char* str, int len)
         ret.valid = 0;
         return ret;
       }
+
+      // maybe limit the query data
```

Comment:
Could we get a more detailed comment here? Ie, why and how.

Reply:
See what you think of the revision.